cmd/slinky/config/config_test.go

```diff
@@ -349,6 +349,52 @@ func TestReadOracleConfigWithOverrides(t *testing.T) {
 		require.Equal(t, expectedConfig.UpdateInterval, cfg.UpdateInterval)
 		require.Equal(t, expectedConfig.Metrics.PrometheusServerAddress, cfg.Metrics.PrometheusServerAddress)
 	})
+
+	t.Run("overriding a nonexistent provider via config fails", func(t *testing.T) {
+		// create a temp file in the current directory
+		tmpfile, err := os.CreateTemp("", "slinky-config-*.json")
+		require.NoError(t, err)
+
+		defer os.Remove(tmpfile.Name())
+
+		overrides := fmt.Sprintf(`
+		{
+			"updateInterval": "%s",
+			"metrics": {
+				"prometheusServerAddress": "%s"
+			},
+			"providers": {
+				"doesNotExist": {
+					"api": {
+						"endpoints": [
+							{
+								"url": "%s"
+							},
+							{
+								"url": "%s",
+								"authentication": {
+									"apiKey": "%s",
+									"apiKeyHeader": "%s"
+								}
+							}
+						]
+					}
+				}
+			}
+		}
+		`,
+			updateIntervalOverride,
+			prometheusServerOverride,
+			raydium.DefaultAPIConfig.Endpoints[0].URL,
+			endpointOverride.URL,
+			endpointOverride.Authentication.APIKey,
+			endpointOverride.Authentication.APIKeyHeader,
+		)
+		tmpfile.Write([]byte(overrides))
+
+		_, err = cmdconfig.ReadOracleConfigWithOverrides(tmpfile.Name(), marketmap.Name)
+		require.Error(t, err)
```

Comment:
you could do something like `require.ErrorContains(t, err, "overridden key doesNotExist does not correspond to a known provider")`

just to make this test more confident that the error we expect is the one returned